### ISSUE TITLE
PDF Attachments are corrupted into blank pages

### ISSUE THREAD

**Issue by EmileSpecs** [zammad/zammad#2507]
### Infos:

* Used Zammad version: 2.9.x
* Installation method (source, package, ..): package
* Operating system: Ubuntu 16.04.6
* Database + version: mysql  Ver 14.14 Distrib 5.7.25
* Elasticsearch version: 5.6.15
* Browser + version: Chrome 72.0.3626.119

### Expected behavior:

Downloaded PDF file should have content and not be empty (blank).

### Actual behavior:

Certain PDF attachments are blank when downloaded from Zammad. When I log into the mail server and view the attachment in the web based email client or on a desktop email client, the attachment is not blank.

So the assumption is that somehow the PDF is being corrupted either when Zammad downloads the attachment from the mail server, or it's corrupted when the user downloads the attachments from Zammad to their computer.

I'm using the file system to save attachments.

### Steps to reproduce the behavior:

None / unknown

Yes I'm sure this is a bug and no feature request or a general question.

Interestingly one of the PDF's in question (attached) is 199 bytes smaller when downloaded from Zammad than when downloaded from email client.

Where is this going wrong, and how to fix it?

P.S. Production log shows no errors related to this as far as I can tell.

[Tax Invoice  INX4183 (from email).PDF](https://github.com/zammad/zammad/files/2936143/Tax.Invoice.INX4183.from.email.PDF)
[Tax Invoice INX4183 (from Zammad).PDF](https://github.com/zammad/zammad/files/2936144/Tax.Invoice.INX4183.from.Zammad.PDF)

**Comment by SummerStorm** [comment]
Hi, there. I just tried sending your provided PDF file to Zammad and it worked fine. Can you please try sending the file to your own Zammad instance twice to confirm that it indeed occurs every time with the same file? Did this occur on only one PDF file or did it occur on a certain percentage of your PDF files?

**Comment by EmileSpecs** [comment]
Hi @SummerStorm, thanks for the quick response.

I've also now sent it to Zammad via email and it also opens just fine. Very strange.
This happens with PDF's from two clients in particular.

What doesn't make sense to me however, is that the file works when I open it from an email client but does not when it's opened in Zammad (when received from the client's email address).

Thus when received from the client email and downloaded in Zammad, the file is 199bytes smaller (as you can see from the files I've attached) and then opens as a blank PDF.

So in conclusion, when the file is sent specifically from the client's email, it somehow get's corrupted along the way. When the file is downloaded from an email client and sent from another email address to Zammad, the file does not get corrupted.

I have no idea how to troubleshoot this!

**Comment by MrGeneration** [comment]
Can you please download the RAW-EML-File from the article with the attachment and check if the attached PDF inside the eml-file is fine? If it is, then the issue is caused somewhere within Zammad, if it's blank already at this state, it's most proberbly a communication problem between Zammad and your mail system (or your mail system broke things, but if you fetched the "original" mail from the exact same mailbox, that wouldn't make sense).

**Comment by EmileSpecs** [comment]
I've downloaded the email, see attached.

It's using a strange encoding method ("quoted-printable"):
Content-Type: application/pdf;
	name="Tax Invoice  INX4183.PDF"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: attachment;
	filename="Tax Invoice  INX4183.PDF"

Email clients seem to handle this encoding fine, but Zammad does not seem to like it.
This is probably the issue?

You'll notice there are two PDF's in the email, the base64 encoded PDF opens correctly from Zammad, the quoted-printable encoded one does not. It seems to me like it's not parsing that PDF's data correctly.

[FW Tax Invoice  INX4183.zip](https://github.com/zammad/zammad/files/2936355/FW.Tax.Invoice.INX4183.zip)

**Comment by SummerStorm** [comment]
Thanks for the raw email! We can reproduce this issue now. It's indeed a problem with Zammad's handling of quoted-printable encoded emails.

**Comment by EmileSpecs** [comment]
Thanks @SummerStorm and @MrGeneration for the confirmation.

Approximately how long would it take to fix a bug like this?

**Comment by MrGeneration** [comment]
Hey @EmileSpecs 

this issue is on our list. I'm afraid that I currently can't give you an approximate fixe date.
We currently focus on issues that affect more users (that doesn't mean, that this issue isn't important!).

Sorry about the waiting time.

**Comment by SummerStorm** [comment]
Fixed by 30d3ce2b88c9bbbbd6800f120a7fdaac5893e900.

**Comment by EmileSpecs** [comment]
Hi @SummerStorm 
I'm afraid this has not been resolved, unless it's not been pushed to the latest version which I have installed.
Can you verify whether this fix is present in the latest version which I've updated to.
If it has, I'm still getting varying degrees of corrupted PDF's. Some are completely blank, others have some content but are incomplete.

**Comment by MrGeneration** [comment]
This has been pushed to develop and is not living in the stable yet.
If you want to test this, please use the develop (beware, quite big changes ahead).

If you're on stable, please wait for the next release.
